### PR TITLE
feat(vcpu): vcpu hotplug support in runtime-rs

### PR DESCRIPTION
**Pull request by Ji-Xinyou**
When a container is created or started from a paused state, there should be vcpu number changing correspondingly. This patch allows runtime to update the vcpu number correctly.

Fixes: #5030
Signed-off-by: Ji-Xinyou <jerryji0414@outlook.com>

### CODE REVIEW

**Comment by Ji-Xinyou** [comment]
Since this patch is based on dbs support of hotplug and corresponding api, we need to merge this only after the dbs upcall & hotplug patches are merged.

**Comment by Ji-Xinyou** [comment]
> /test

do not test this, this requires dbs patch to be merged.....

File: src/runtime-rs/crates/resource/src/cpu_mem/mod.rs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+/// Cpu information wrapper
+#[derive(Default, Debug, Clone, Copy)]
+pub struct CpuResource {
+    /// The default vcpu # when boot
+    default_vcpu: i32,
+
+    /// Current number of vCPUs
+    /// when boot, the number is set to *default_vcpus*
+    current_vcpu: i32,
+    // TODO(shuoyu.ys): cpuset mapping relation
```
Comment (studychao) [inline src/runtime-rs/crates/resource/src/cpu_mem/mod.rs:27]: Please add the issue number for every TODO

**Comment by studychao** [comment]
Close this since I 'll continue @Ji-Xinyou 's work in thie PR #6009

**Pull request closed without merging**